### PR TITLE
Add anding of facets functionality

### PR DESCRIPTION
**Pull request by jonescc**

### CODE REVIEW

**Comment by dnahodil** [comment]
I'll check this out

**Comment by anguss00** [comment]
We have a problem with tests here that isn't replicated in jasmine in the browser

**Comment by dnahodil** [comment]
Yeah, I noticed that too. I'll have a look into that but Travis is not building at the moment.

File: src/test/javascript/portal/search/FacetFilterPanelSpec.js [hunk 1]
```diff
@@ -15,23 +16,57 @@ describe("Portal.search.FacetFilterPanel", function() {
...
-            expect(searcher.addDrilldownFilter).toHaveBeenCalledWith(selectedDrilldownPath);
+            filterPanel.setSelectedDrilldown(0, ['Measured Parameter', 'Salinity']);
+            expect(searcher.search).toHaveBeenCalled();
+            expect(searcher.filterCount()).toEqual(1);
+            expect(searcher.hasDrilldown(['Measured Parameter', 'Salinity'])).toEqual(true);
+        });
+
+        it('calls search with all selected drilldowns', function() {
+            filterPanel.setSelectedDrilldown(0, ['Measured Parameter', 'Salinity']);
+            filterPanel.tools.plus.dom.click();
```
Comment (anguss00) [inline src/test/javascript/portal/search/FacetFilterPanelSpec.js]: Travis seems to get upset on this click function... investigating

**Comment by jonescc** [comment]
Jasmine htmlunit tests fixed.

**Comment by dnahodil** [comment]
Looks good, @jonescc

**Pull request merged**